logos/benchmarks/benchmark_logos.py

```
@@ -2652,6 +2653,65 @@ def _set_calibration_window_enabled(
+    Detect that drift up front and repair the ``is_active`` flags directly in
+    the database (same host access the benchmark already uses for compose
+    management). A missing key or a non-admin role cannot be self-repaired and
+    raises immediately, BEFORE any multi-hour run starts.
+    """
+    if not re.fullmatch(r"[A-Za-z0-9_-]+", logos_key or ""):
+        raise RuntimeError("LOGOS key has unexpected characters; refusing to embed it in SQL.")
+    sudo = "sudo " if use_sudo else ""
+    repair_sql = (
+        f"UPDATE users u SET is_active = true FROM api_keys ak "
+        f"WHERE ak.user_id = u.id AND ak.key_value = '{logos_key}' AND NOT u.is_active; "
+        f"UPDATE api_keys SET is_active = true WHERE key_value = '{logos_key}' AND NOT is_active;"
```

Ruff (0.15.20)

[error] Possible SQL injection vector through string-based query construction

(S608)